readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -839,6 +841,241 @@ def test_command_oom_kill(self):
+            'commonmark==0.5.4',
+            'recommonmark==0.4.0',
+            'mkdocs==0.15.0',
+        ]
+        requirements = self.base_requirements + requirements_mkdocs
+        args = self.pip_install_args + requirements
+        self.build_env_mock.run.assert_called_once_with(
+            *args, bin_path=mock.ANY
+        )
+
+    def test_install_user_requirements(self):
+        """
+        If a projects does not specify a requirements file,
+        RTD will choose one automatically.
+
+        First by searching under the docs/ directory and then under the root.
+        The files can be named as:
+
+        - ``pip_requirements.txt``
+        - ``requirements.txt``
+        """
+        self.build_env_mock.project = self.project_sphinx
+        self.build_env_mock.version = self.version_sphinx
+        python_env = Virtualenv(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock
+        )
+
+        checkout_path = python_env.checkout_path
+        docs_requirements = os.path.join(
+            checkout_path, 'docs', 'requirements.txt'
+        )
+        root_requirements = os.path.join(
+            checkout_path, 'requirements.txt'
+        )
+        paths = {
+            os.path.join(checkout_path, 'docs'): True,
+        }
+        args = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '--exists-action=w',
+            '--cache-dir',
+            mock.ANY,  # cache path
+            'requirements_file'
+        ]
+
+        # One requirements file on the docs/ dir
+        # should be installed
+        paths[docs_requirements] = True
+        paths[root_requirements] = False
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(docs_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        # One requirements file on the root dir
+        # should be installed
+        paths[docs_requirements] = False
+        paths[root_requirements] = True
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(root_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        # Two requirements files on the root and  docs/ dirs
+        # the one on docs/ should be installed
+        paths[docs_requirements] = True
+        paths[root_requirements] = True
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(docs_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        # No requirements file
+        # no requirements should be installed
+        self.build_env_mock.run.reset_mock()
```

Comment:
With the last update of the mock lib, the `assert_not_called()` was counting the previous calls too, so I have to reset it here.